### ISSUE TITLE
JobResult views raise FieldError if linked model class lacks a Name attribute

### ISSUE THREAD

**Issue by glennmatthews** [nautobot/nautobot#482]
### Environment
* Python version:
* Nautobot version: 1.0.1

### Steps to Reproduce
1. Instantiate a JobResult with its `obj_type` referencing a ContentType for a model (built-in or plugin-provided) that lacks a `name` attribute
2. Attempt to view either `/extras/job-results/` or `/extras/job-results/<uuid of JobResult record>`

### Expected Behavior

View loads without errors but is possibly missing a link to the associated object.

### Observed Behavior

Unhandled `FieldError` due to querying `model_class.objects.get(name=...)` when `model_class` doesn't have a `name` field.

**Comment by jathanism** [comment]
Weird, why would a content-type not have a name? That's a concrete field!

**Comment by glennmatthews** [comment]
The latter. I tried to explain this a bit better in the linked PR but basically `JobResult` has a `obj_type` (which is a foreign key to `ContentType`), a `name` (which is currently used, depending on the `obj_type` value, to either look up a (virtual) `Job` object by its `class_path`, or a `GitRepository` record by its `name`), and a `job_id` (which is required to be unique, but is otherwise unused). 

This problem arises when (as I discovered in working on developing a plugin) the `obj_type` references a model class that does not have a `name` field, as the current code is basically "Is the `obj_type = Job`? If so look up a Job whose `class_path` matches `name`; else, look up a record of type `obj_type` whose `name` matches `name`", which throws a `FieldError`.
